### ISSUE TITLE
Mixture models ($MIXTURE): discrete latent subpopulations with covariate-dependent mixing

### ISSUE THREAD

**Issue by roninsightrx** [FeRx-NLME/ferx-core#977]
## Goal

Support NONMEM `$MIXTURE`-style discrete latent subpopulations: each subject belongs to one of K unobserved classes with mixing probabilities `p_ik`. Class-conditional parameters are estimated jointly with the mixing probabilities, and each subject receives a posterior class membership (NONMEM `MIXEST`/`PMIX` analogue).

FeRx has **no** mixture-model support today. (The only "mixture" in the codebase is the unrelated defensive-mixture IMP/IMPMAP importance-sampling proposal.)

## Locked scope (design decisions)

| Decision | Choice |
|---|---|
| **Mixing probabilities** | Covariate-dependent `p(k)` (full): per-subject expression over theta + covariates, softmax/logistic-normalized. Constant `p(k)` is the degenerate case. Matches NONMEM `$MIX` (arbitrary abbreviated code over `THETA` + data items). |
| **What differs per class** | Fully separate param sets: independent θ (via `MIXNUM` branching, user picks which are shared), Ω, Σ (via per-class declaration). |
| **Estimators (v1)** | FOCE/FOCEI first. Other methods (SAEM/IMP/Laplace/AGQ/Bayes) error clearly until wired. |
| **DSL** | New `[mixture]` block + reserved read-only `MIXNUM` index usable in `[individual_parameters]`. |

**Level reconciliation:** `MIXNUM` `if/else` in `[individual_parameters]` handles **θ** differences (and lets the user share what they want). Ω and Σ are population params, not `[individual_parameters]` expressions, so per-class Ω/Σ needs explicit per-class declaration in `[mixture]`. The plan supports the fully-separate case that way.

## Math

Per-subject marginal, class-conditional Laplace:

```
p_ik = softmax_k( g_k(θ, covariates_i) )          Σ_k p_ik = 1
L_i  = Σ_k  p_ik · L_ik(θ_k, Ω_k, Σ_k)
OFV  = −2 Σ_i log Σ_k p_ik · exp(−nll_ik)          ← log-sum-exp for stability
```

`nll_ik` = existing `foce_subject_nll` with class-k params + class-k EBE. Cost ≈ **K×** the inner loop (K EBE searches per subject per outer iteration). Posterior membership `PMIX_ik = p_ik·exp(−nll_ik) / Σ_j (…)`; `MIXEST_i = argmax_k PMIX_ik`.

## DSL design

```
[parameters]
  theta TVCL1(1.0); theta TVCL2(3.0)   # class-specific CL typical values
  theta TVV(30.0)                       # shared
  theta MIXL(0.0); theta BWT(0.0)       # mixing logit intercept + covariate slope
  omega ETA_CL ~ 0.1                     # class 1 IIV (base)
  sigma PROP ~ 0.05                      # class 1 residual (base)

[mixture]
  nsub = 2
  logit(1) = MIXL + BWT*WT               # p(1)=σ(logit); p(2)=1−p(1). K>2 → softmax over logit(1..K-1)
  omega(2) ETA_CL ~ 0.4                  # class-2 Ω override (optional; else shared)
  sigma(2) PROP ~ 0.12                   # class-2 Σ override (optional; else shared)

[individual_parameters]
  CL = if MIXNUM == 1 : TVCL1 * exp(ETA_CL) else TVCL2 * exp(ETA_CL)
  V  = TVV
```

Rules:
- `MIXNUM` is a reserved read-only index (1..K) usable anywhere in `[individual_parameters]`.
- `logit(k)` / `p(k)` accept the same expression grammar as individual params (theta + covariates).
- Omitted `omega(k)` / `sigma(k)` ⇒ shared with base.
- Parser rejects: `MIXNUM` write, `p(k)` referencing `eta`, `nsub<2`, missing branch coverage, `MIXNUM` in a non-mixture model.

## Implementation map (verified seams)

| Concern | Location | Change |
|---|---|---|
| Objective fold | `stats/likelihood.rs:2123` `foce_population_nll` | per-subject term → mixture log-sum-exp over K class NLLs (new `foce_subject_mixture_nll`) |
| Per-subject class NLL | `stats/likelihood.rs:895` `foce_subject_nll` | called K times, once per class param set |
| Inner EBE | `estimation/inner_optimizer.rs:715` `find_ebe`, `:3160` `run_inner_loop_warm` | run per (subject × class) with class-k Ω prior + Σ; store `Vec<EbeResult>`, warm-start per class |
| Mixing-coef packing | `estimation/parameterization.rs:41/88/190` | logit **coefficients** ride the existing θ vector (identity-packed, `theta_packs_log` at `:29` already allows negatives); no new probability transform. New packed segments for per-class Ω/Σ overrides (mirror `:57`/`:68`) |
| DSL declaration | `parser/model_parser.rs` `parse_parameters` `:9565`, block dispatch `:1043` | new `[mixture]` block; new `Expression::MixNum`; reuse expr parser for `logit(k)`, omega/sigma regexes for overrides |
| Per-subject eval switching | `parser/model_parser.rs:11471` `eval_statements_indexed_with_stack` | thread class index in so existing `if/else` (`Statement::If :11665`) branches on `MIXNUM` |
| Data model | `types.rs` | `MixtureSpec` on `CompiledModel`; `mixture: Option<MixtureParams>` on `ModelParameters` (per-class Ω/Σ overrides); `pk_param_fn` closure gains a class index |
| Sensitivities | `sens/` | `∂OFV/∂θ` through log-sum-exp + softmax via `Dual2`; add `*_g_dual_matches_fd` parity for the combiner; FD-route + test any out-of-scope class |
| Per-subject output | `types.rs:3713` `SubjectResult` | add `mix_probs: Vec<f64>` (PMIX), `mix_est: usize`; sdtab cols `PMIX1..PMIXK`, `MIXEST`; ipred/cwres at the subject's `MIXEST` class |
| AIC/BIC param count | `api/fit.rs:885,1514` | counts all class θ/Ω/Σ + mixing coefficients automatically via `packed_fixed_mask` once packing is extended |

**Reuse win:** the DSL already has `if / else if / else` + `Expression::Conditional` (`:11549`) and categorical-covariate branching that already converges (`tests/categorical_convergence.rs`). Class-switching of typical values reuses this. New work = (1) mixing-probability expressions, (2) `MIXNUM` index, (3) per-class Ω/Σ, (4) marginalization in the objective.

## Phasing

1. Parser + data model + validation (`fit` errors "mixture not wired").
2. Packing + per-class param plumbing + round-trip tests.
3. Objective (K-fold log-sum-exp) + K× inner loop → FOCE convergence.
4. FOCEI + sensitivities + `Dual2`-vs-FD parity.
5. Postfit outputs (PMIX/MIXEST/sdtab) + AIC/BIC.
6. Covariance/SE over mixing coefficients; NONMEM validation; docs.

## Testing & validation (per CLAUDE.md)

- **Tier-1:** parser (block, `MIXNUM`, validation errors); packing round-trip; log-sum-exp combiner; `Dual2`-vs-FD parity for the mixture gradient.
- **Tier-2:** `fit()` returns after a few iters on a 2-class model.
- **Tier-3 (slow-gated):** full convergence on a bimodal-CL dataset.
- **NONMEM `$MIXTURE` comparison (required):** θ_k, Ω_k, p(k), OFV, per-subject `MIXEST` vs an equivalent NONMEM run — in `docs/estimation/mixture.qmd` or the PR.
- **Docs:** new `docs/estimation/mixture.qmd` + `_quarto.yml` entry; `MIXNUM` in `individual-parameters.qmd`; `[mixture]` keys in `fit-options.qmd`; CHANGELOG `Added` entry.
- **Cross-repo:** new `pub` API (`MixtureSpec`, `FitResult`/`SubjectResult` fields) ⇒ follow-up `ferx-r` PR + `Cargo.lock` bump (`tools/update-ferx-core-lock.sh`).

## Risks / open questions

- **Identifiability / label switching** — classic mixture pitfall; needs multi-start + a label-ordering convention (order classes by a reference θ). Flag non-convergence loudly.
- **K× cost** — inner loop is already the bottleneck; parallelize across (subject × class).
- **SE of mixing coefficients** — FD-of-OFV Hessian (`estimation/covariance.rs`) must include the logit thetas and handle softmax curvature.
- **Empty classes** — a class no subject prefers → `p(k)→0` degeneracy; add a floor + warning.

**Comment by TeunP** [comment]
Closing — delivered by PR #980 (merged 2026-08-18): `[mixture]` block, `MIXNUM`, FOCE/FOCEI/SAEM/Bayes, covariate-dependent mixing, `PMIX`/`MIXEST` in sdtab and the `.fitrx`, NONMEM-anchored in `tests/mixture_nonmem.rs`. Anything left from the plan (SEs over mixing coefficients, if still wanted) should be a fresh, narrow issue rather than this open plan.
